src/ngraph/frontend/onnx_import/utils/reshape.cpp

```diff
@@ -56,36 +61,100 @@ namespace ngraph
                     ngraph::AxisVector{input_order},
                     ngraph::Shape{first_dim_size, last_dim_size});
             }
-        } // namespace utils
 
-        std::shared_ptr<ngraph::Node> reorder_axes(const std::shared_ptr<ngraph::Node>& node,
-                                                   std::vector<size_t> axes_order = {})
-        {
-            ngraph::Shape out_shape = node->get_shape();
-            if (axes_order.empty())
+            AxisVector get_default_axis_vector(std::size_t data_shape_size, std::size_t start_value)
             {
-                axes_order.resize(out_shape.size());
-                std::iota(std::begin(axes_order), std::end(axes_order), 0);
+                AxisVector axis_vector(data_shape_size);
+                std::iota(std::begin(axis_vector), std::end(axis_vector), start_value);
+                return axis_vector;
             }
-            else
+
+            std::vector<std::size_t> infer_dimensions(std::string node_name,
+                                                      std::vector<std::size_t> input_shape,
+                                                      std::vector<std::size_t> output_shape)
             {
-                for (auto i = 0; i < axes_order.size(); ++i)
+                // If an output dimension is equal to zero its actual value is copied from the input
+                // shape argument.
+                for (std::size_t idx = 0; idx < output_shape.size(); ++idx)
+                {
+                    if (output_shape.at(idx) == 0)
+                    {
+                        try
```

Comment:
I would argue that this construct make no sense since:
`for (std::size_t idx = 0; idx < output_shape.size(); ++idx)`
I'm iterating over `output_shape` vector, thus make sure that accessing its value won't raise an error (at least I hope it won't). However maybe using exceptions mechanism is an overkill for this case.